src/server/db_slice.cc

```diff
@@ -495,6 +495,44 @@ bool DbSlice::Del(DbIndex db_ind, PrimeIterator it) {
   return true;
 }
 
+void DbSlice::FlushSlotsFb(const SlotSet& slot_ids) {
+  // Slot deletion can take time as it traverses all the database, hence it runs in fiber.
+  // We want to flush all the data of a slot that was added till the time the call to FlushSlotsFb
+  // was made. Therefore we delete slots entries with version < next_version
+  uint64_t next_version = NextVersion();
+
+  std::string tmp;
+  auto del_entry_cb = [&](PrimeTable::iterator it) {
+    std::string_view key = it->first.GetSlice(&tmp);
+    SlotId sid = ClusterConfig::KeySlot(key);
+    if (slot_ids.contains(sid) && it.GetVersion() < next_version) {
+      PerformDeletion(it, shard_owner(), db_arr_[0].get());
+    }
+    return true;
+  };
+
+  PrimeTable* pt = &db_arr_[0]->prime;
+  PrimeTable::Cursor cursor;
+  uint64_t i = 0;
+  do {
+    PrimeTable::Cursor next = pt->Traverse(cursor, del_entry_cb);
+    ++i;
+    cursor = next;
+    if (i % 100 == 0) {
+      ThisFiber::Yield();
+    }
+
+  } while (cursor);
+  mi_heap_collect(ServerState::tlocal()->data_heap(), true);
+}
+
+void DbSlice::FlushSlots(const SlotSet&& slot_ids) {
```

Comment:
you do need `const` I guess

Reply:
Yes, a `const T&&` is something that should not exist.
If you want to take ownership over a parameter, just take it by value (i.e. `FlushSlots(SlotSet slot_ids)`). That will make it possible to call this function both with rvalues (which will be efficient and will duck an unnecessary copy), or with an lvalue (meaning just a regular variable), which will copy but will not be less efficient than accepting a `const T&`.
By accepting a `T&&` argument, you are forcing the caller to pass in a temporary, i.e. this code won't work:
```cpp
SlotSet slots;
// ...
FlushSlots(slots)
```
So only use that convention to *force* callers not to copy.